### PR TITLE
Add interface to show all books for each status

### LINKED ISSUE #210: Add book and status detail interface for each status
Include at least the barcode and druid in a table linked from the status page.

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Fixes #210 

This adds an interface to navigate the various statuses that books will be in. It includes pagination as the ready list will be very long for a while, and the others will grow as well.

A follow up PR will add individual book detail and links to `retry` and `retry all` from the error status pages.

Screen Shot 2020-01-31 at 1 39 28 PM

Pagination:
Screen Shot 2020-01-31 at 1 39 36 PM

Screen Shot 2020-01-31 at 9 15 51 AM

Screen Shot 2020-01-31 at 9 15 57 AM
 

## Was the documentation (README, DevOpsDocs, wiki) updated?

N/A

### CODE REVIEW

File: app/controllers/statuses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      { id: 'req-succeeded', label: 'Process Request Succeeded', status: Book::PROCESS_REQ_SUCCEEDED },
+      { id: 'req-failed', label: 'Process Request Failed', status: Book::PROCESS_REQ_FAILED },
+      { id: 'process-failed', label: 'Process Failed', status: Book::PROCESS_FAILED },
+      { id: 'download-ready', label: 'Download Ready', status: Book::DOWNLOAD_READY },
+      { id: 'download-succeeded', label: 'Download Succeeded', status: Book::DOWNLOAD_SUCCEEDED },
+      { id: 'download-failed', label: 'Download Failed', status: Book::DOWNLOAD_FAILED },
+      { id: 'md5-failed', label: 'MD5 Check Failed', status: Book::MD5_CHECK_FAILED },
+      { id: 'deposit-succeeded', label: 'Deposit Succeeded', status: Book::DEPOSIT_SUCCEEDED },
+      { id: 'deposit-failed', label: 'Deposit Failed', status: Book::DEPOSIT_FAILED },
+      { id: 'success', label: 'Success', status: Book::SUCCESS_REQ_STATUS }
```
Comment (justinlittman) [inline app/controllers/statuses_controller.rb]: This isn't a book status. (It is a status returned by GRIN.)

File: app/views/statuses/_books.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div>
+  <table class="table table-striped">
+  <thead>
+    <tr>
+      <th scope="col">Barcode</th>
+      <th scope="col">Druid</th>
```
Comment (justinlittman) [inline app/views/statuses/_books.html.erb]: Last update (like errors report) would also be helpful.

File: app/controllers/statuses_controller.rb (hunk 1 above)
Comment (aaron-collier) [reply app/controllers/statuses_controller.rb]: Ah, do we want to show that... somewhere? Or just not be concerned with it for now?

File: app/views/statuses/_books.html.erb (hunk 2 above)
Comment (aaron-collier) [reply app/views/statuses/_books.html.erb]: Yeah, that's what I was thinking will go there generally, and druid on the success. This was just a first pass.

**Comment by aaron-collier** [comment]
@jcoyne that makes sense :+1:

File: app/controllers/statuses_controller.rb (hunk 1 above)
Comment (justinlittman) [reply app/controllers/statuses_controller.rb]: Not concerned. It is used in processing the responses from GRIN.

**Pull request merged**